ecommerce-website/src/containers/Products/Products.js

```diff
@@ -0,0 +1,113 @@
+import React,{Component} from 'react';
+import Product from '../../components/Product/Product';
+import styles from './Products.module.css';
+import axios from 'axios';
+import Loader from '../../components/ui/loader/Loder';
+import SearchSort from '../../components/ui/SearchSort/SearchSort';
+
+class Products extends Component{
+    state={
+        products:null,
+        error:false,
+        loading:true,
+        productsbycategory:null,
+        searchProduct:null
+    }
+    componentDidMount(){ 
+        axios.get( 'https://fakestoreapi.com/products' )
+            .then( response => {
+                this.setState({products:response.data,loading:false})
+            } )
+            .catch( error => {
+                this.setState({error:true,loading:false})
+            } );
+    }
+    clickHandler=(id)=>{
+        this.props.history.push(`/products/${id}`)
+    }
+    sortHandler = (e)=>{
```

Comment:
Sorting could have been handled in better way. One way I can think of is through redux.